src/lib/json_source/json_source_client.js

```diff
@@ -0,0 +1,63 @@
+import promises from "../../../public/promises.json";
+
+const JsonSourceClient = () => {
+  return {
+    query({ query, limit, category, id }) {
+      switch (query) {
+        case "GET_PROMISES":
+          return new Promise((resolve, reject) => {
+            if (promises.promises === undefined) {
+              reject(new Error("No promises defined"));
+            }
+            resolve(promises.promises.slice(0, limit));
+          });
+        case "GET_PROMISE":
+          return new Promise((resolve, reject) => {
+            try {
+              const promise = promises.promises.filter(
+                (p) => `${p.id}` === id
+              )[0];
+              return resolve(promise);
+            } catch (e) {
+              return reject(e);
+            }
+          });
+        case "GET_KEY_PROMISES":
+          return new Promise((resolve, reject) => {
+            if (promises.promises === undefined) {
+              reject(new Error("No promises defined"));
+            }
+            const keyPromises = promises.promises.filter(
+              (p) => `${p.key_promise}` === "True"
+            );
```

Comment:
```suggestion
            const keyPromises = promises?.promises?.filter(
              (p) => `${p.key_promise}` === "True"
            ) ?? [];
```

I think it's best we return empty promises whenever we can rather than rejecting Promise or throwing errors.

PS: Any reason why we're defining Promises when we already have the JSON file imported?

Reply:
@kilemensi  for 

>PS: Any reason why we're defining Promises when we already have the JSON file imported?

I'm not sure I fully understand your inquiry. Do you mean doing just 

`import  "../../../public/promises.json";`
  
should be enough?

Reply:
`Promise` as in `new Promise` i.e. going async when we already have everything loaded in memory. We're not doing any network calls, etc. so I don't see Promises adding anything here.

Reply:
Oooh. I see

Reply:
I think it will be useful once we migrate to using URLs for the JSON file source,

Reply:
@kilemensi  Given that the final goal is to load JSON from a server, I think we can maintain this async functionality. What do you think?

Reply:
Not sure @esirK ... Are we going to fetch the JSON file periodically from the URL and store it in the public folder for catching? This means all our processing will be done from local JSON file which can load in memory.

Reply:
Yeah. In that case, we really don't need the async.